src/common/AmazonS3Backend.ts

```diff
@@ -293,4 +293,18 @@ export class AmazonS3Backend implements IBackend {
     });
   }
 
+  /**
+   * Initiate S3 multi-part upload using ManagedUpload
+   */
+  public initMultipartUpload(params: any): Promise<any> {
+    return new Promise((resolve: any, reject: any) => {
+      // add ManagedUpload object to the file
+      params.Body.s3upload = new AWS.S3.ManagedUpload({
```

Comment:
Rather than manually creating a ManagedUpload object, we can use the `this.s3.upload(params, callback)`.  This would also eliminate the need to have a separate function for initializing the multipart upload.

Reply:
Thanks for pointing this out. I've refactored the code so we do not have two functions for initializing and sending files, but I kept the manual creation of ManagedUpload since we still need to define an event handler on the ManagedUpload object for each file which would listen for 'httpUploadProgress' and update the progress bar in dropzone. These changes are found on the 'rev3' branch. Will be closing this PR.